File: backend/app/graphql/types/legal_notice.py
import strawberry
from typing import Optional, List
import uuid
from datetime import datetime


@strawberry.type
class LegalNoticeType:
    id: uuid.UUID
    notice_type: str
    recipient_name: str
    recipient_address: Optional[str]
    sender_name: str
    sender_address: Optional[str]
    sender_email: Optional[str]
    form_data: Optional[str]
    generated_content: Optional[str]
    created_at: datetime
    updated_at: Optional[datetime]


@strawberry.input
class CreateLegalNoticeInput:
    notice_type: str
    recipient_name: str
    recipient_address: Optional[str] = None
    sender_name: str
    sender_address: Optional[str] = None
    sender_email: Optional[str] = None
    form_data: Optional[str] = None
    generated_content: Optional[str] = None


@strawberry.input
class UpdateLegalNoticeInput:
    id: str
    recipient_name: Optional[str] = None
    recipient_address: Optional[str] = None
    sender_name: Optional[str] = None
    sender_address: Optional[str] = None
    sender_email: Optional[str] = None
    form_data: Optional[str] = None
    generated_content: Optional[str] = None

File: frontend/src/graphql/legalNotice.ts
import { gql } from "@apollo/client";

export const GET_LEGAL_NOTICES = gql`
  query GetLegalNotices {
    getLegalNotices {
      id
      notice_type
      recipient_name
      recipient_address
      sender_name
      sender_address
      sender_email
      form_data
      generated_content
      created_at
      updated_at
    }
  }
`;

export const GET_LEGAL_NOTICE = gql`
  query GetLegalNotice($noticeId: String!) {
    getLegalNotice(noticeId: $noticeId) {
      id
      notice_type
      recipient_name
      recipient_address
      sender_name
      sender_address
      sender_email
      form_data
      generated_content
      created_at
      updated_at
    }
  }
`;

export const CREATE_LEGAL_NOTICE = gql`
  mutation CreateLegalNotice($input: CreateLegalNoticeInput!) {
    createLegalNotice(input: $input) {
      id
      notice_type
      recipient_name
      recipient_address
      sender_name
      sender_address
      sender_email
      form_data
      generated_content
      created_at
      updated_at
    }
  }
`;

export const UPDATE_LEGAL_NOTICE = gql`
  mutation UpdateLegalNotice($input: UpdateLegalNoticeInput!) {
    updateLegalNotice(input: $input) {
      id
      notice_type
      recipient_name
      recipient_address
      sender_name
      sender_address
      sender_email
      form_data
      generated_content
      created_at
      updated_at
    }
  }
`;

export const DELETE_LEGAL_NOTICE = gql`
  mutation DeleteLegalNotice($noticeId: String!) {
    deleteLegalNotice(noticeId: $noticeId) {
      success
      message
    }
  }
`;
